### ISSUE TITLE
is hot reload possible?

### ISSUE THREAD

**Issue by longzhiri** [fwsGonzo/rvscript#10]
Hello! If I have a rather large script project, and I want to hot reload a single script file, just like updating a module in Lua, is that possible?

**Comment by fwsGonzo** [comment]
No, it would only be possible if we could, for example, embed TCC (a C compiler), which then generated RISC-V from C code. We'd no longer be scripting in C++, but at least we could use languages that transpile to C and  of course C itself. Unfortunately, I could never figure out how to generate 64-bit RISC-V with TCC on-the-fly using a cross compilation configuration.

I know there are some projects that embed (or even download) Zig, and Zig can compile both itself and C code. That also opens up doors, but it will make it hard for multi-platform projects. It would definitely be a cool thing to try, though.

Bottom line is: Currently no!

**Comment by fwsGonzo** [comment]
Reading your question again, I will say that for example in this repository, if you change a single script file, only that file will be recompiled when you build. All my build scripts use incremental compilation. I also use ccache in order to reduce build times even further. The result is that the iteration speed is quite OK, despite having to build a program after each change.

I don't know if it can compete with a Lua hot-reload. But it would be possible to detect a source change and rebuild the RISC-V binary in the background with a simple shell script that uses inotify.
Example: https://gist.github.com/mpapi/4656389
